hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/DynamicConfigurationAwareConfig.java

```diff
@@ -1227,12 +1230,17 @@ public Config setExternalDataStoreConfigs(Map<String, ExternalDataStoreConfig> e
 
     @Override
     public Config addExternalDataStoreConfig(ExternalDataStoreConfig externalDataStoreConfig) {
-        throw new UnsupportedOperationException("Unsupported operation");
+        boolean staticConfigDoesNotExist = checkStaticConfigDoesNotExist(staticConfig.getExternalDataStoreConfigs(),
+                externalDataStoreConfig.getName(), externalDataStoreConfig);
+        if (staticConfigDoesNotExist) {
+            configurationService.broadcastConfig(externalDataStoreConfig);
+        }
+        return this;
     }
 
     @Override
     public ExternalDataStoreConfig getExternalDataStoreConfig(String name) {
-        return staticConfig.getExternalDataStoreConfig(name);
+        return (ExternalDataStoreConfig) configSearcher.getConfig(name, name, supplierFor(ExternalDataStoreConfig.class));
     }
 
     @Override
```

Comment:
Not related to this line but as far as I understand, `findExternalDataStoreConfig` method should also take a look at the dynamically added configurations. Looking at other `findXXX` methods, they do it like that.

Reply:
thanks, unified: https://github.com/hazelcast/hazelcast/pull/21930/commits/512f7ae4770c2071f8f7e63a2935682fe8d49eea